R/theme.R

```diff
@@ -455,6 +469,14 @@ theme <- function(line,
     }
     elements$legend.text.align <- NULL
   }
+  if (is.numeric(elements[["legend.position"]])) {
+    deprecate_soft0(
+      "3.5.0", I("A numeric `legend.position` argument in `theme()`"),
+      "theme(legend.position.inside)"
+    )
+    elements$legend.position.inside <- elements$legend.position
+    elements$legend.position <- "inside"
+  }
```

Comment:
Can you elaborate on this change? I agree in principle but with the historic bagage I think we need a clear reason for deprecating a mode of a theme element

Reply:
Mostly because there are several places where `legend.position` is used to make decisions about directions or justifications and it is a pain to add an `if (is.numeric(legend.position)` block every time. The numeric position itself is only used in one place relatively late in the guide building process, so converting it from numeric to `"inside"` once at the beginning is not an option. It just seemed cleaner to separate it.

Reply:
Sorry I just remembered a clearer reason why I wanted to separate numeric position from position options. If `position = "inside"` at the guide level but at the theme level `legend.position` is not `"inside"` (or numeric), there wouldn't be a mechanism to control the placement of guides inside the panel (other than defaulting to the justification).

Reply:
oh, so `position` can't be a numeric inside the `guide_*()` specification, only in the theme. I guess that makes sense that we want all "inside" guides in the same spot

Reply:
Yes that's right, there is just one guide box for the inside position. I guess we *could* allow numeric positions in guide functions, but I wasn't prepared to deal with that chaos quite yet.